thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -3196,8 +3369,16 @@ bool CBucket::addRow(const void *row, unsigned hashValue)
         {
             // Even if not adding, check HT for dedupping purposes upfront
             unsigned htPos = hashValue % htRows->queryMaxRows();
-            if (htRows->lookupRow(htPos, row))
-                return false; // dedupped
+            unsigned matchedHtPos = htRows->lookupRow(htPos, row);
+            if (matchedHtPos != UINT_MAX)
+            {
+                const void * oldrow = htRows->queryRow(matchedHtPos);
+                if (!keepBest || keepBestCompare->docompare(oldrow,row) <= 0)
+                    return false; // dedupped
+
+                ReleaseRoxieRow(htRows->getRowClear(matchedHtPos));
```

Comment:
Could you add a comment above this line, along the lines that this is removing non-best row?